src/js/models/layerstructure.js

```diff
@@ -822,10 +822,13 @@ define(function (require, exports, module) {
             if (Number.isInteger(replace)) {
                 // if explicitly replacing, then replace by current ID
                 replaceLayer = this.byID(replace);
-            } else {
-                // otherwise, replace the selected layer
-                var selectedLayers = document.layers.selected;
-                replaceLayer = selectedLayers && selectedLayers.size === 1 && selectedLayers.first();
+            } else if (this.selected && this.selected.size === 1) {
+                // otherwise, replace the selected layer if it has the same index with the new layer.
+                var selectedLayer = this.selected.first(),
+                    selectedLayerIndex = this.indexOf(selectedLayer),
+                    newLayerIndex = descriptors[0].itemIndex;
+                    
+                replaceLayer = selectedLayerIndex === newLayerIndex && selectedLayer;
```

Comment:
The change here depends on the descriptor attribute `itemIndex` from PS, so it should be reliable to do a replacement when the new layer has the same index with the background layer ? Here are some scenarios I tested and passed:

* select bg layer, draw a rectangular, bg layer replaced
* selecte multiple layers (including bg layer), draw a rectangular, bg layer NOT replaced
* select bg layer, drag asset (except for Ai asset, which will create a new layer) from libraries, bg layer replaced
* deselect bg layer, draw a rectangular, bg layer NOT replaced

Please let me know if I miss anything here.